backend/src/packages/articles/article.repository.ts

```diff
@@ -10,8 +10,23 @@ class ArticleRepository implements IRepository {
     this.articleModel = articleModel;
   }
 
-  public findAll(): Promise<unknown[]> {
-    return Promise.resolve([]);
+  public async findAll(): Promise<ArticleEntity[]> {
+    const articles = await this.articleModel
+      .query()
+      .orderBy('publishedAt', 'desc')
+      .withGraphFetched('userDetails');
+
+    return articles.map((article) => ArticleEntity.initializeWithUser(article));
+  }
+
+  public async findOwn(id: number): Promise<ArticleEntity[]> {
+    const articles = await this.articleModel
+      .query()
+      .where('userId', id)
+      .orderBy('publishedAt', 'desc')
+      .execute();
+
+    return articles.map((article) => ArticleEntity.initialize(article));
   }
```

Comment:
You do not need to create duplicate `findOwn` repository method - just use extend `findAll` method to take optional filtering values (in future there would be more filters applied)

Reply:
Do I need to make eager loading for author both in all articles and user's articles?